server/services/invoice.service.js

```diff
@@ -1,22 +1,45 @@
 const TransacStatDao = require("../data_access_layer/daos/transac_stat.dao");
 const InvoiceAffectDao = require("../data_access_layer/daos/invoice_affect.dao");
-const NameQualityDao = require("../data_access_layer/daos/name_quality.dao");
 const ClientDao = require("../data_access_layer/daos/name.dao");
 const ClientGradingDao = require("../data_access_layer/daos/client_grading.dao")
+const CountryDao = require("../data_access_layer/daos/country.dao")
+const NameQualityDao = require("../data_access_layer/daos/name_quality.dao");
+
+
+
+
+exports.getAverages = async (startDateStr, endDateStr,  employeeId = undefined, countryCode = undefined) => {
+
+    const startYear = parseInt(startDateStr.split('-')[0]);
+    let startMonth = parseInt(startDateStr.split('-')[1]);
+    const endYear = parseInt(endDateStr.split('-')[0]);
+    const endMonth = parseInt(endDateStr.split('-')[1]);
 
-let clientIDList = [];
+    // make list of yearMonth [201911,202001,202002,...] to select dues
+    let yearMonthList = [];
+    for (let y = startYear; y <= endYear; y++) {
+        if (y != startYear) startMonth = 1;
+        for (let m = startMonth; m <= 12; m++) {
+            if (y == endYear && m > endMonth) break;
+            let yearMonthStr = y.toString();
+            if (m < 10) yearMonthStr += '0';
+            yearMonthStr += m.toString();
+            yearMonthList.push(parseInt(yearMonthStr));
+        }
+    }
```

Comment:
You can remove all of this except the `let yearMonthList = [];` because we are already making the yearMonthList using the `getYearMonth` function at line 53